ft_badchannel.m

```diff
@@ -0,0 +1,239 @@
+%   cfg.feedback      = 'yes' or 'no', whether to show an image of the neighbour values (default = 'no')
+%
+% The 'neighcorrel' and 'neighstdratio' methods implement the bad channel detection
+% (more or less) according to the paper "Adding dynamics to the Human Connectome
+% Project with MEG", Larson-Prior et al. https://doi.org/10.1016/j.neuroimage.2013.05.056.
+%
+% Most methods compute a scalar value for each channel that can simply be
+% thresholded. The NEIGHBCORR and NEIGHBSTDRATIO compute a vector with a value for
+% each of the neighbour of a channel. The cfg.nbdetect option allows you to specify
+% whether you want to flag the channel as bad in case 'all' of its neighbours exceed
+% the threshold, if 'most' exceed the threshold, or if 'any' of them exceeds the
+% threshold. Note that when you specify 'any', then all channels neighbouring a bad
+% channel will also be marked as bad, since they all have at least one bad neighbour.
+% You can also specify 'median', in which case the threshold is applied to the median
+% value over neighbours.
+%
+% See also FT_CHANNELREPAIR, FT_REJECTVISUAL
+
+% Copyright (C) 2021, Robert Oostenveld
+%
+% This file is part of FieldTrip, see http://www.fieldtriptoolbox.org
+% for the documentation and details.
+%
+%    FieldTrip is free software: you can redistribute it and/or modify
+%    it under the terms of the GNU General Public License as published by
+%    the Free Software Foundation, either version 3 of the License, or
+%    (at your option) any later version.
+%
+%    FieldTrip is distributed in the hope that it will be useful,
+%    but WITHOUT ANY WARRANTY; without even the implied warranty of
+%    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+%    GNU General Public License for more details.
+%
+%    You should have received a copy of the GNU General Public License
+%    along with FieldTrip. If not, see <http://www.gnu.org/licenses/>.
+%
+% $Id$
+
+% these are used by the ft_preamble/ft_postamble function and scripts
+ft_revision = '$Id$';
+ft_nargin   = nargin;
+ft_nargout  = nargout;
+
+% do the general setup of the function
+ft_defaults
+ft_preamble init
+ft_preamble debug
+ft_preamble loadvar data
+ft_preamble provenance
+ft_preamble trackconfig
+
+% the ft_abort variable is set to true or false in ft_preamble_init
+if ft_abort
+  return
+end
+
+% check if the input data is valid for this function
+data = ft_checkdata(data, 'datatype', 'raw', 'feedback', 'yes');
+
+% ensure that the configuration is consistent
+cfg = ft_checkconfig(cfg, 'required', 'method');
+
+% ensure that the preproc specific options are located in the cfg.preproc substructure
+cfg = ft_checkconfig(cfg, 'createsubcfg',  {'preproc'});
+
+% set the defaults
+cfg.nbdetect      = ft_getopt(cfg, 'nbdetect', 'all');
+cfg.feedback      = ft_getopt(cfg, 'feedback', 'no');
+cfg.thresholdside = ft_getopt(cfg, 'thresholdside', []); % the default depends on cfg.method, see below
+
+if isempty(cfg.thresholdside)
```

Comment:
You probably thought about this, but are there use cases where cfg.thresholdside's value is meaningfully non-default, as a function of cfg.method? In other words, does this argument need to be configureable by the user?

Reply:
you are right. I initially made it configurable to postpone thinking about it, but later I indeed identified that the default is obvious. There is only one obscure case left where it is ambiguous. I will remove it.